lua/plenary/functional.lua

```diff
@@ -38,4 +38,21 @@ f.all = function(f, iterable)
   return true
 end
 
+function f.make_selector(n)
+  return function(...)
+    local x = select(n, ...)
+    return x
+  end
+end
+
+f.first = f.make_selector(1)
+f.second = f.make_selector(2)
+f.third = f.make_selector(3)
+
+function f.last(...)
+  local length = select('#', ...)
+  local x = select(length, ...)
```

Comment:
Of course could do `local x = select(select('#', ...), ...)` instead. Probably more readable with the intermediate value though, so feel free to ignore this ;)